From e6240cbb33b6ef31332259d2d2484da57681790a Mon Sep 17 00:00:00 2001
From: Ganesh Vilas Chaudhari <quic_ganeshvi@quicinc.com>
Date: Tue, 19 Sep 2023 14:12:33 +0530
Subject: [PATCH 1/2] Disable AOSP module for targets no Telephony

  - CallLogBackup
  - CellBroadcastReceiver
  - CarrierConfig
  - CarrierDefaultApp
  - com.android.cellbroadcast
  - Dialer
  - EmergencyInfo
  - ONS
  - Stk
  - TeleService

Change-Id: I4d90f0a54d5cbb35d28b6bbc459d3b736fe2496d
CRs-Fixed: 3620652
---
 target/product/generic_system.mk    | 8 ++++++--
 target/product/handheld_system.mk   | 8 ++++++--
 target/product/telephony_product.mk | 2 ++
 target/product/telephony_system.mk  | 6 +++++-
 4 files changed, 19 insertions(+), 5 deletions(-)

diff --git a/target/product/generic_system.mk b/target/product/generic_system.mk
index a072e50..54953a9 100644
--- a/target/product/generic_system.mk
+++ b/target/product/generic_system.mk
@@ -32,10 +32,14 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     LiveWallpapersPicker \
     PartnerBookmarksProvider \
-    preinstalled-packages-platform-generic-system.xml \
-    Stk \
     Tag \
 
+ifneq ($(TARGET_NO_TELEPHONY), true)
+PRODUCT_PACKAGES += \
+    preinstalled-packages-platform-generic-system.xml \
+    Stk
+endif #TARGET_NO_TELEPHONY
+
 ifeq ($(RELEASE_AVATAR_PICKER_APP),true)
   PRODUCT_PACKAGES += \
     AvatarPicker
diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index 38e6cf4..3adfb8d 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -56,7 +56,6 @@ PRODUCT_PACKAGES += \
     KeyChain \
     librs_jni \
     ManagedProvisioning \
-    MmsService \
     MtpService \
     MusicFX \
     PacProcessor \
@@ -68,12 +67,17 @@ PRODUCT_PACKAGES += \
     SimAppDialog \
     Telecom \
     TelephonyProvider \
-    TeleService \
     Traceur \
     UserDictionaryProvider \
     VpnDialogs \
     vr \
 
+ifneq ($(TARGET_NO_TELEPHONY), true)
+PRODUCT_PACKAGES += \
+  MmsService \
+  TeleService
+endif #TARGET_NO_TELEPHONY
+
 PRODUCT_PACKAGES += $(RELEASE_PACKAGE_VIRTUAL_CAMERA)
 # Set virtual_camera_service_enabled soong config variable based on the
 # RELEASE_PACKAGE_VIRTUAL_CAMERA build. virtual_camera_service_enabled soong config
diff --git a/target/product/telephony_product.mk b/target/product/telephony_product.mk
index 0705511..32d9ab3 100644
--- a/target/product/telephony_product.mk
+++ b/target/product/telephony_product.mk
@@ -18,6 +18,8 @@
 # hardware, and install to the product partition.
 
 # /product packages
+ifneq ($(TARGET_NO_TELEPHONY), true)
 PRODUCT_PACKAGES += \
     ImsServiceEntitlement \
     preinstalled-packages-platform-telephony-product.xml
+endif #TARGET_NO_TELEPHONY
diff --git a/target/product/telephony_system.mk b/target/product/telephony_system.mk
index 81e7aa2..80b6c65 100644
--- a/target/product/telephony_system.mk
+++ b/target/product/telephony_system.mk
@@ -21,10 +21,14 @@
 TARGET_USES_QCOM_BSP_ATEL := true
 
 PRODUCT_PACKAGES := \
+
+ifneq ($(TARGET_NO_TELEPHONY), true)
+PRODUCT_PACKAGES += \
     ONS \
     CarrierDefaultApp \
     CallLogBackup \
     com.android.cellbroadcast \
-    CellBroadcastLegacyApp \
+    CellBroadcastLegacyApp
+endif #TARGET_NO_TELEPHONY
 
 PRODUCT_COPY_FILES := \
-- 
2.47.1

